src/data-access/repositories/interfaces/core-device-repository.ts

```diff
@@ -0,0 +1,5 @@
+import { CoreDevice } from '../../../model/entities/core-device'
+
+export interface CoreDeviceRepository {
```

Comment:
This interface shouldn't exist. 
We only have one abstraction for a device repository (aka. `DeviceRepository`).
Both `MongoDeviceRepository` and `MongoCoreDeviceRepository` should implement that interface.